### ISSUE TITLE
Native source code changes should trigger a warning/error on full restart

### ISSUE THREAD

**Issue by fmatosqg** [flutter/flutter#16493]
When people change commits either due to merging branches, cherry picking etc there's the potential to bring in changes in files such as manifest, source code, plist etc. 

While the flutter engine can survive those changes they will tend to display inconsistent behaviour if dart code needs to communicate with native. And the developer may be completely unaware that native files have changed.

I'd suggest something very obvious, ranging from refusing to update to trigger a full build or flashing something very obvious on the device's screen. Although I think the last alternative is not a good one for a number of reasons: message shown while screen is locked, devices without screens (Android Things doesn't require it) or the trend with foldable displays.

In short,
- Full native build if any native file changed
- Also the term "full restart" is misleading since it doesn't seem to touch native build tools.
- A warning on hot reload should be added as well, it would be nice to have something like when state is not initialised from scratch.

Stretch goal, tap into AS instant run. But that can get hairy since I still don't fully trust some things it does, and it wont bring benefits in half the cases.

**Comment by zoechi** [comment]
Do you think this is still an issue?
I think native code is updated for hot-reload or hot-restart.

**Comment by zoechi** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!
Thanks for your contribution.
